### DISCUSSION TITLE
Extension gets stuck at secret storage step

### DISCUSSION THREAD

**Discussion by theshubhamjoshi** [microsoft/vscode-discussions#945] [Extension Development QnA]
Hey there!

We are running into an issue where the extension code execution gets stuck on the secret storage line for some reason. This happens when we try to store the user credentials via the secret storage API. We are doing it like this:

`await extensionContext.secrets.store(this._authTokenKey, JSON.stringify(accountData));`

We tried putting in some logs into the entire execution flow, and we observed that the execution never moves past beyond this point in the CI environment. Strangely there are no errors logged as well. We tried some of the flags which came up when we searched for similar issues with secret storage.

We tried the following flags while launching the VSCode IDE:
```
--password-store="gnome"
--use-inmemory-secretstorage
```

But these flag don’t seem to fix the issue. Need some help in identifying the issue here :slightly_smiling_face:

GitHub Actions runner specs
`Ubuntu Latest (22.04) · 4-cores · 16 GB RAM · 150 GB SSD`

Here is the renderer.log file
[renderer.log](https://github.com/microsoft/vscode-discussions/files/13474470/renderer.log)

**Comment by r3m0t** [comment]
Try adding `--verbose` to the VSCode flags.

**Reply by theshubhamjoshi** [reply]
Thanks for this tip @r3m0t. I was able to obtain some detailed logs by adding the verbose flag. Looking at the renderer.log file, it seems that the execution is stuck on this point in kind of an infinite loop. I am seeing this log repeatedly till the end of file

`2023-11-27 10:48:39.531 [trace] [mainThreadSecretState] Getting password for postman.postman-for-vscode-preview extension:  postman-for-vscode-preview.authToken`

Any ideas as to what the issue might be?

**Reply by r3m0t** [reply]
The full logs would be better.. Did you also check other VS Code extensions on GitHub Actions?

**Reply by theshubhamjoshi** [reply]
Not really didnt try any other extensions but here is the renderer.log file

Here is the renderer.log file
[renderer.log](https://github.com/microsoft/vscode-discussions/files/13474470/renderer.log)

@r3m0t

**Comment by olguzzar** [comment]
Hi @theshubhamjoshi, it's been a while since there's been any activity on this discussion. Are you still having this issue?

**Comment by abrilurena** [comment]
Hi @theshubhamjoshi! it's been a while and just wanted to check in and see if you still need help with this?
